src/compiler/checker.ts

```diff
@@ -18923,7 +18923,11 @@ namespace ts {
                     generalizedSourceType = getTypeNameForErrorDisplay(generalizedSource);
                 }
 
-                if (target.flags & TypeFlags.TypeParameter && target !== markerSuperTypeForCheck && target !== markerSubTypeForCheck) {
+                const targetFlags = target.flags & TypeFlags.IndexedAccess && !(source.flags & TypeFlags.IndexedAccess) ?

```

Comment:
Maybe add a comment so this can be understood at a glance.

Reply:
I added a comment now